en/micro-integrator/docs/references/connectors/gmail-connector/gmail-connector-example.md

```diff
@@ -0,0 +1,104 @@
+            </gmail.init>
+            <gmail.listAllMails>
+                <includeSpamTrash>false</includeSpamTrash>
+                <maxResults>20</maxResults>
+                <q>is:unread label:customers</q>
+            </gmail.listAllMails>
+            <iterate expression="json-eval($.messages)">
+                <target>
+                    <sequence>
+                        <sequence key="reply"/>
+                    </sequence>
+                </target>
+            </iterate>
+            <respond/>
+        </inSequence>
+        <outSequence/>
+        <faultSequence/>
+    </resource>
+</api>
+```
+
+5. Right click on the created ESB Solution Project and select, -> **New** -> **Sequence** to create the defined sequence called **reply**. 
+
+6. Provide the Sequence name as **reply**. You can go to the source view of the xml configuration file of the API and copy the following configuration. 
+```
+<?xml version="1.0" encoding="UTF-8"?>
+<sequence name="reply" trace="disable" xmlns="http://ws.apache.org/ns/synapse">
+    <property expression="json-eval($.id)" name="msgId" scope="default" type="STRING"/>
+    <gmail.getAccessTokenFromRefreshToken>
+        <clientId></clientId>
+        <clientSecret></clientSecret>
+        <refreshToken></refreshToken>
+    </gmail.getAccessTokenFromRefreshToken>
+    <gmail.readMail>
+        <id>{$ctx:msgId}</id>
+    </gmail.readMail>
+    <property expression="json-eval($.payload.headers[6].value)" name="response" scope="default" type="STRING"/>
+    <log level="custom">
+        <property expression="get-property('response')" name="response1"/>
+    </log>
+    <gmail.getAccessTokenFromRefreshToken>
+        <clientId></clientId>
+        <clientSecret></clientSecret>
+        <refreshToken></refreshToken>
+    </gmail.getAccessTokenFromRefreshToken>
+    <gmail.sendMail>
+        <to>{$ctx:response}</to>
+        <subject>Best of Europe - 6 Countries in 9 Days</subject>
+        <from>isurumuy@gmail.com</from>
+        <messageBody>Thank you for your valuable feedback.</messageBody>
+    </gmail.sendMail>
+</sequence>
+```
+7. In the Rest API and in the Sequence, provide your obtained **Client ID**, **Client Secret**, **Access Token** and **Refresh Token** accordingly. The **userID** should be your gmail address. 
+
+8. Follow these steps to export the artifacts. 
+{!references/connectors/exporting-artifacts.md !}
+
+## Deployment
+Follow these steps to deploy the exported CApp in the Enterprise Integrator Runtime. 
+{!references/connectors/deploy-capp.md!}
+
+## Testing
+Invoke the API as shown below using the curl command. Curl Application can be downloaded from [here](https://curl.haxx.se/download.html).
+
+```
+  curl -H "Content-Type: application/json" --request GET http://localhost:8290/sendmails
+```
+
+The senders should receive an email with a subject of "Best of Europe — 6 Countries in 9 Days", and a body of "Thank you for your valuable feedback."
+
```

Comment:
Add a section similar to the following (customize it for Gmail connector):

## What's Next

* You can deploy and run your project on [Docker](../../../setup/installation/run_in_docker.md) or [Kubernetes](../../../setup/installation/run_in_kubernetes.md).
* To customize this example for your own scenario, see [File Connector Configuration](../file-connector-config.md) documentation.

Reply:
Done